cmd/istio_ca/main.go

```diff
@@ -38,47 +38,79 @@ const (
 	namespaceKey = "NAMESPACE"
 )
 
+type cliOptions struct {
+	certChainFile   string
+	signingCertFile string
+	signingKeyFile  string
+	rootCertFile    string
+
+	namespace      string
+	kubeConfigFile string
+
+	selfSignedCA    bool
+	selfSignedCAOrg string
+
+	caCertTTL time.Duration
+	certTTL   time.Duration
+}
+
 var (
-	certChainFile   = flag.String("cert-chain", "", "Speicifies path to the certificate chain file")
-	signingCertFile = flag.String("signing-cert", "", "Specifies path to the CA signing certificate file")
-	signingKeyFile  = flag.String("signing-key", "", "Specifies path to the CA signing key file")
-	rootCertFile    = flag.String("root-cert", "", "Specifies path to the root certificate file")
-	namespace       = flag.String("namespace", "",
+	opts cliOptions
+
+	rootCmd = &cobra.Command{
+		Run: func(cmd *cobra.Command, args []string) {
+			runCA()
+		},
+	}
+)
+
+func init() {
+	flags := rootCmd.Flags()
+
+	flags.StringVar(&opts.certChainFile, "cert-chain", "", "Speicifies path to the certificate chain file")
+	flags.StringVar(&opts.signingCertFile, "signing-cert", "", "Specifies path to the CA signing certificate file")
+	flags.StringVar(&opts.signingKeyFile, "signing-key", "", "Specifies path to the CA signing key file")
+	flags.StringVar(&opts.rootCertFile, "root-cert", "", "Specifies path to the root certificate file")
+
+	flags.StringVar(&opts.namespace, "namespace", "",
 		"Select a namespace for the CA to listen to. If unspecified, Istio CA tries to use the ${"+namespaceKey+"} "+
 			"environment variable. If neither is set, Istio CA listens to all namespaces.")
-
-	kubeConfigFile = flag.String("kube-config", "",
+	flags.StringVar(&opts.kubeConfigFile, "kube-config", "",
 		"Specifies path to kubeconfig file. This must be specified when not running inside a Kubernetes pod.")
 
-	selfSignedCA = flag.Bool("self-signed-ca", false,
+	flags.BoolVar(&opts.selfSignedCA, "self-signed-ca", false,
 		"Indicates whether to use auto-generated self-signed CA certificate. "+
 			"When set to true, the '-ca-cert' and '-ca-key' options are ignored.")
```

Comment:
This looks wrong. Somehow we no longer has `-ca-cert` and `-ca-key` defined. I'll look into it more